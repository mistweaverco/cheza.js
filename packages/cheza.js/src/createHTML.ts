import type { ChezaDataStore } from '../types'
import UIIcons from '../icons/sprite/icons.svg'
import { createButton } from './createButton'
import { createSVG } from './createSVG'

const createRootSVG = (): void => {
  if (document.getElementById('chezajs-default-ui-icons') === null) {
    const c = document.createElement('div')
    c.id = 'chezajs-default-ui-icons'
    c.innerHTML = UIIcons
    document.body.appendChild(c)
  }
}

export const createHTML = (dataStore: ChezaDataStore): HTMLDivElement => {
  createRootSVG()
  const root = document.createElement('div')
  root.classList.add('cheza')
  dataStore.videoElement.parentElement?.insertBefore(root, dataStore.videoElement)
  root.appendChild(dataStore.videoElement)
  const controls = ['cheza__controls--top', 'cheza__controls--bottom']
  controls.forEach((cls) => {
    const ctl = document.createElement('div')
    ctl.classList.add(cls)
    root.appendChild(ctl)
  })
  const buttons = [
    {
      ariaLabel: 'Play',
      title: 'Play',
      svg: createSVG('play', {}),
      on: [
        {
          name: 'click',
          callback: () => {
            void dataStore.videoElement.play()
          }
        }
      ]
    }
  ]
  buttons.forEach((btn) => {
    const button = createButton(btn)
    root.querySelector('.cheza__controls--bottom')?.appendChild(button)
  })

  return root
}
