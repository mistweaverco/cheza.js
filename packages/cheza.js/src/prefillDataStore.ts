import type { ChezaDataStore } from '../types'

export const prefillDataStore = (videoElement: HTMLVideoElement): ChezaDataStore => {
  const rootElement = document.createElement('div')
  const progressContainer = document.createElement('div')
  const progress = document.createElement('div')
  const controlsTop = document.createElement('div')
  const controlsBottom = document.createElement('div')
  const videoContainer = document.createElement('div')
  const loadingSpinner = document.createElement('div')
  const activePlugins = new Map()
  return {
    rootElement,
    videoElement,
    progressContainer,
    progress,
    controlsTop,
    controlsBottom,
    videoContainer,
    loadingSpinner,
    activePlugins
  }
}
