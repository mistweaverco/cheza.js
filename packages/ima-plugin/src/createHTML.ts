import type { ChezaIMADataStore } from '../types'
import { createButton, type ChezaButtonOpts } from './createButton'
import { GetBottomButtonOptsDirection, getBottomButtonOpts } from './getBottomButtonOpts'

export const createHTML = (dataStore: ChezaIMADataStore): void => {
  const defaultUIElement = dataStore.defaultUIElement as HTMLDivElement
  const uiRoot = dataStore.ui.root
  const overlay = dataStore.ui.overlay
  overlay.classList.add('overlay')
  uiRoot.classList.add('ima-ui', 'hidden')
  defaultUIElement.after(uiRoot)
  dataStore.ui.adContainer.classList.add('ad-container')
  uiRoot.appendChild(dataStore.ui.adContainer)
  const loadingSpinnerAnimation = document.createElement('div')
  loadingSpinnerAnimation.classList.add('animation')
  dataStore.ui.loadingSpinner.classList.add('loading-spinner', 'hidden')
  dataStore.ui.loadingSpinner.appendChild(loadingSpinnerAnimation)
  overlay.appendChild(dataStore.ui.loadingSpinner)
  dataStore.ui.controlsTop.classList.add('controls-top')
  dataStore.ui.controlsBottom.classList.add('controls-bottom')
  overlay.appendChild(dataStore.ui.controlsTop)
  overlay.appendChild(dataStore.ui.controlsBottom)
  uiRoot.appendChild(overlay)

  getBottomButtonOpts(dataStore, GetBottomButtonOptsDirection.LEFT).forEach((buttonOpts: ChezaButtonOpts) => {
    dataStore.ui.controlsBottom.appendChild(createButton(buttonOpts))
  })
  const timeLabel = document.createElement('div')
  timeLabel.classList.add('time-label')
  timeLabel.innerText = 'Ad'
  dataStore.ui.controlsBottom.appendChild(timeLabel)
  getBottomButtonOpts(dataStore, GetBottomButtonOptsDirection.RIGHT).forEach((buttonOpts: ChezaButtonOpts) => {
    dataStore.ui.controlsBottom.appendChild(createButton(buttonOpts))
  })
}
