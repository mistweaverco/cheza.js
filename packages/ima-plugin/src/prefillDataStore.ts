import type { ChezaIMADataStore } from '../types'

export const prefillDataStore = (): ChezaIMADataStore => {
  const root = document.createElement('div')
  const adContainer = document.createElement('div')
  const controlsTop = document.createElement('div')
  const controlsBottom = document.createElement('div')
  const loadingSpinner = document.createElement('div')
  return {
    defaultUIElement: undefined,
    videoElement: undefined,
    imaVideoElement: undefined,
    totalAds: 0,
    adPosition: 0,
    adsManager: undefined,
    adsLoader: undefined,
    adDisplayContainer: undefined,
    rootElement: undefined,
    adsInitialized: false,
    google: undefined,
    adsLoaded: false,
    opts: {},
    ui: {
      root,
      overlay: document.createElement('div'),
      adContainer,
      controlsTop,
      controlsBottom,
      loadingSpinner
    }
  }
}
