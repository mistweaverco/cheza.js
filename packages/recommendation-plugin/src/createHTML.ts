import type { ChezaRecommendationDataStore } from '../types'

export const createHTML = (dataStore: ChezaRecommendationDataStore): void => {
  const defaultUIElement = dataStore.defaultUIElement as HTMLDivElement
  const uiRoot = dataStore.ui.root
  uiRoot.classList.add('recommendation-ui', 'hidden')
  defaultUIElement.after(uiRoot)
  dataStore.ui.content.classList.add('content')
  uiRoot.appendChild(dataStore.ui.content)
}
