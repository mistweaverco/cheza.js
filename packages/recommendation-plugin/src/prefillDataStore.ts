import type { ChezaRecommendationDataStore } from '../types'

export const prefillDataStore = (): ChezaRecommendationDataStore => {
  const root = document.createElement('div')
  const content = document.createElement('div')
  return {
    defaultUIElement: undefined,
    videoElement: undefined,
    rootElement: undefined,
    intervalTicker: null,
    opts: {},
    ui: {
      root,
      content
    }
  }
}
