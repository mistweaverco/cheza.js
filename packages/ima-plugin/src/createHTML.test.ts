import { createHTML } from './createHTML'
import { prefillDataStore } from './prefillDataStore'

describe('createHTML', () => {
  it('should match the snapshot', () => {
    const dataStore = prefillDataStore()
    dataStore.videoElement = document.createElement('video')
    dataStore.rootElement = document.createElement('div')
    dataStore.defaultUIElement = document.createElement('div')
    createHTML(dataStore)
    expect(dataStore.ui.root).toMatchSnapshot()
  })
})
