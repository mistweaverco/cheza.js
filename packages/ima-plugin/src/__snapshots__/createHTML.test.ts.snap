// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createHTML should match the snapshot 1`] = `
<div
  class="ima-ui hidden"
>
  <div
    class="ad-container"
  />
  <div
    class="overlay"
  >
    <div
      class="loading-spinner hidden"
    >
      <div
        class="animation"
      />
    </div>
    <div
      class="controls-top"
    />
    <div
      class="controls-bottom"
    >
      <button
        aria-label="Play"
        class="button button-play"
        title="Play"
        type="button"
      >
        <svg
          focusable="false"
          role="presentation"
        >
          <use
            xlink:href="#cheza-play"
          />
        </svg>
      </button>
      <button
        aria-label="Pause"
        class="button button-pause hidden"
        title="Pause"
        type="button"
      >
        <svg
          focusable="false"
          role="presentation"
        >
          <use
            xlink:href="#cheza-pause"
          />
        </svg>
      </button>
      <button
        aria-label="Mute"
        class="button button-volume"
        title="Mute"
        type="button"
      >
        <svg
          focusable="false"
          role="presentation"
        >
          <use
            xlink:href="#cheza-volume"
          />
        </svg>
      </button>
      <button
        aria-label="Unmute"
        class="button button-muted hidden"
        title="Unmute"
        type="button"
      >
        <svg
          focusable="false"
          role="presentation"
        >
          <use
            xlink:href="#cheza-muted"
          />
        </svg>
      </button>
      <div
        class="time-label"
      />
      <button
        aria-label="Enter Fullscreen"
        class="button button-enter-fullscreen"
        title="Enter Fullscreen"
        type="button"
      >
        <svg
          focusable="false"
          role="presentation"
        >
          <use
            xlink:href="#cheza-enter-fullscreen"
          />
        </svg>
      </button>
      <button
        aria-label="Exit Fullscreen"
        class="button button-exit-fullscreen hidden"
        title="Exit Fullscreen"
        type="button"
      >
        <svg
          focusable="false"
          role="presentation"
        >
          <use
            xlink:href="#cheza-exit-fullscreen"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
